pkg/operator/controller/instance_controller.go

```diff
@@ -0,0 +1,742 @@
+}
+
+// setControlplane sets up the controlplane deployment.
+func (r *InstanceReconciler) setControlplane(ctx context.Context) error {
+	cpDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ControlPlaneName,
+			Namespace: r.clNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": ControlPlaneName},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": ControlPlaneName},
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "ca",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "cl-fabric",
+								},
+							},
+						},
+						{
+							Name: "tls",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: ControlPlaneName,
+								},
+							},
+						},
+						{
+							Name: ControlPlaneName,
+							VolumeSource: corev1.VolumeSource{
+								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+									ClaimName: ControlPlaneName,
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:            ControlPlaneName,
+							Image:           r.instance.Spec.ContainerRegistry + ControlPlaneName + ":" + r.instance.Spec.ImageTag,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Args:            []string{"--log-level", r.instance.Spec.LogLevel, "--platform", "k8s"},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: cpapi.ListenPort,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "ca",
+									MountPath: cpapp.CAFile,
+									SubPath:   "ca",
+									ReadOnly:  true,
+								},
+								{
+									Name:      "tls",
+									MountPath: cpapp.CertificateFile,
+									SubPath:   "cert",
+									ReadOnly:  true,
+								},
+								{
+									Name:      "tls",
+									MountPath: cpapp.KeyFile,
+									SubPath:   "key",
+									ReadOnly:  true,
+								},
+								{
+									Name:      ControlPlaneName,
+									MountPath: filepath.Dir(cpapp.StoreFile),
+								},
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name: "CL-NAMESPACE",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "metadata.namespace",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return r.createOrUpdateResource(ctx, cpDeployment)
+}
+
+// setDataplane sets up the dataplane deployment.
+func (r *InstanceReconciler) setDataplane(ctx context.Context) error {
+	DataplaneImage := DataPlaneName
+	if r.instance.Spec.DataPlane.Type == api.DataplaneTypeGo {
+		DataplaneImage = GoDataPlaneName
+	}
+
+	dpDeployment := &appsv1.Deployment{
```

Comment:
nit: Consider adding a function that sets all of the common fields in the two deployments and then overriding specific items in each. Might keep code shorter.
Or maybe just the Volume stuff can be extracted?